nvm.sh

```diff
@@ -1907,6 +1908,58 @@ nvm_download_artifact() {
   nvm_echo "${TARBALL}"
 }
 
+# args: tarball path, NVM_OS var
+nvm_extract_tar_artifact() {
+  local tar_compression_flag
+  local TARBALL
+  local TAR_COMMAND_SUFFIX
+  local TMPDIR
+  local NVM_OS
+  local tar
+
+  TARBALL="${1}"
+  NVM_OS="${2}"
+
+  tar='tar'
+  if [ "${NVM_OS}" = 'aix' ]; then
+    tar='gtar'
+  fi
+
+  tar_compression_flag='z'
+
+  if nvm_supports_xz "${VERSION}" && [ "_$NVM_OS" != "_openbsd" ]; then
+    tar_compression_flag='J'
+  fi
+
+  if [ "_$NVM_OS" = "_openbsd" ] && nvm_has xz; then
```

Comment:
if the version doesn't support xz, then it shouldn't try to use xz on it.
```suggestion
  if [ "_$NVM_OS" = "_openbsd" ] && nvm_has xz && nvm_supports_xz "${VERSION}"; then
```